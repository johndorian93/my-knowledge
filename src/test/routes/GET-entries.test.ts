import request from 'supertest';

import { createTestServer } from "../testServer";

describe('GET /entries endpoint', () => {
    let closeServer: () => void;

    afterEach(() => {
        if (closeServer) {
            closeServer();
        }
    });

    it('returns list of entries stored in a database', async () => {
        const testServer = await createTestServer();

        const server = testServer.server;
        closeServer = testServer.closeServer;

        const createEntryResult = await request(server)
            .post('/entries')
            .send({
                id: 1,
                content: 'Test content 1',
            });

        expect(createEntryResult.statusCode).toBe(201);
        expect(createEntryResult.body).toMatchSnapshot();

        const getEntryResult = await request(server)
            .get(`/entries/${createEntryResult.body.id}`);

        expect(getEntryResult.statusCode).toBe(200);
        expect(getEntryResult.body).toMatchSnapshot();

        const getAllEntriesResult = await request(server)
            .get('/entries');

        expect(getAllEntriesResult.statusCode).toBe(200);
        expect(getAllEntriesResult.body).toMatchSnapshot();
    });
});
