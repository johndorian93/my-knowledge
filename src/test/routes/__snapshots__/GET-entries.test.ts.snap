// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /entries endpoint returns list of entries stored in a database 1`] = `
Object {
  "content": "Test content 1",
  "id": 1,
  "structuredContent": Object {
    "type": "text",
    "value": "Test content 1",
  },
}
`;

exports[`GET /entries endpoint returns list of entries stored in a database 2`] = `
Object {
  "content": "Test content 1",
  "id": 1,
  "structuredContent": Object {
    "type": "text",
    "value": "Test content 1",
  },
}
`;

exports[`GET /entries endpoint returns list of entries stored in a database 3`] = `
Array [
  Object {
    "content": "Test content 1",
    "id": 1,
    "structuredContent": Object {
      "type": "text",
      "value": "Test content 1",
    },
  },
]
`;
